### PR TITLE
Removed babel polyfill

### PR DESCRIPTION
**Pull request by JasonStoltz**
## Description

`@babel/polyfill` was a dependency in this project. This was a pretty crazy headache, as it was a runtime dependency (as opposed to a build time, dev dependencies) that actually goes out and Polyfills a number of global APIs, like `Promise`, `Map`, etc, which is very intrusive and there is a lot of potential for conflict there. It's especially bad since this is a library, and you would *not* expect a library to do that.

In Babel's docs, they actually outline two methods for handling these globals. The first is the Polyfill, which they advise against for the libraries. The second, is https://babeljs.io/docs/en/babel-plugin-transform-runtime, which they DO recommend for libraries.

The approach the plugin takes is to replace globals like "Promise.all" etc with helpers from the `babel-runtime` package. This is much safer for libraries as it's not actually shimming anything globally.

Initially, I attempted to remove `@babel/polyfill` and not replace it, but it is required since we use `async` functions, which require a regenerate runtime.

Long story short, I replaced `@babel/polyfill` with `@babel/plugin-transform-runtime`

## List of changes
- Removed `@babel/polyfill`
- Added `@babel/plugin-transform-runtime`

## Associated Github Issues
https://github.com/elastic/search-ui/issues/278
https://github.com/elastic/search-ui/issues/290

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -122,7 +122,7 @@ It can often be useful to publish a change and test it before doing a real live
 option for this.
 
 1. Check out pr branch
-2. `npx lerna publish --canary` - Publishing with canary generates a unique new version number, publishes it to npm, then updates the `canary` tag in npm to point to that new version.
+2. `npx lerna publish --canary --preid canary` - Publishing with canary generates a unique new version number, publishes it to npm, then updates the `canary` tag in npm to point to that new version.
```
Comment (constancecchen) [inline CONTRIBUTING.md]: This command is starting to become a bit of a mouthful - do we want to create a package.json `scripts` shorthand for this? something like `canary`, so devs can just type `npm run canary` or `yarn canary` instead?

Comment (JasonStoltz) [reply CONTRIBUTING.md]: Yes, that is a great idea. I can do that in a follow up PR.
